packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -69,12 +70,22 @@ export class BlockList extends Component {
 			// do replace here
 			this.props.replaceBlock( this.props.selectedBlockClientId, newBlock );
 		} else {
-			const indexAfterSelected = this.props.selectedBlockOrder + 1;
-			const insertionIndex = indexAfterSelected || this.props.blockCount;
-			this.props.insertBlock( newBlock, insertionIndex );
+			this.props.insertBlock( newBlock, this.getNewBlockInsertionIndex() );
 		}
 	}
 
+	getNewBlockInsertionIndex() {
+		if ( this.props.isPostTitleSelected ) {
+			// if post title selected, insert at top of post
+			return 0;
+		} else if ( this.props.selectedBlockOrder === -1 ) {
+			// if no block selected, insert at end of post
+			return this.blockCount;
```

Comment:
This probably should be:
```suggestion
			return this.props.blockCount;
```
Funny thing is that this still works since index is allowed to be undefined, and the default is to add the new block at the end:
https://github.com/WordPress/gutenberg/blob/933150fe21f61524a0a371bfbc0897084a571036/packages/block-editor/src/store/reducer.js#L752

Anyway I think is still good to send the proper index in case this change in the future.

Reply:
You're right. Thanks for catching that!